### PR TITLE
OSDOCS10847: Added placementGroupPartition field in machine set AWS existing placement group

### PR DESCRIPTION
**Pull request by subhtk**
Version(s): 4.17+

Issue: https://issues.redhat.com/browse/OSDOCS-10847

Link to docs preview: https://81645--ocpdocs-pr.netlify.app/openshift-enterprise/latest/machine_management/creating_machinesets/creating-machineset-aws.html#machineset-aws-existing-placement-group_creating-machineset-aws

QE review:
- [x] QE has approved this change.

SME review:
- [x] SME has approved this change.

Additional information:

### CODE REVIEW

File: modules/machineset-aws-existing-placement-group.adoc [hunk 1]
```diff
@@ -55,13 +55,15 @@ spec:
             availabilityZone: <zone> # <3>
             region: <region> # <4>
           placementGroupName: <placement_group> # <5>
+          placementGroupPartition: <placement partition number> # <6>
```
Comment (chiragkyal) [inline modules/machineset-aws-existing-placement-group.adoc:58]:
```suggestion
          placementGroupPartition: <placement_group_partition_number> # <6>
```

File: modules/machineset-aws-existing-placement-group.adoc [hunk 2]
```diff
@@ -55,13 +55,15 @@ spec:
...
 # ...
 ----
 <1> Specify an instance type that link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html#efa-instance-types[supports EFAs].
 <2> Specify the `EFA` network interface type.
 <3> Specify the zone, for example, `us-east-1a`.
 <4> Specify the region, for example, `us-east-1`.
 <5> Specify the name of the existing AWS placement group to deploy machines in.
+<6> Specify the partition number of the existing AWS placement group to deploys machines in.
```
Comment (chiragkyal) [inline modules/machineset-aws-existing-placement-group.adoc:66]: Can we somehow write that this field is optional?

File: modules/machineset-aws-existing-placement-group.adoc [hunk 3]
```diff
@@ -55,13 +55,15 @@ spec:
...
 # ...
 ----
 <1> Specify an instance type that link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html#efa-instance-types[supports EFAs].
 <2> Specify the `EFA` network interface type.
 <3> Specify the zone, for example, `us-east-1a`.
 <4> Specify the region, for example, `us-east-1`.
 <5> Specify the name of the existing AWS placement group to deploy machines in.
+<6> Specify the partition number of the existing AWS placement group to deploys machines in. 
 
 .Verification
```
Comment (chiragkyal) [inline modules/machineset-aws-existing-placement-group.adoc:68]: One line can be added to the `Verification` for `placementGroupPartition` like `placementGroupName`

File: modules/machineset-aws-existing-placement-group.adoc
Comment (chiragkyal) [inline modules/machineset-aws-existing-placement-group.adoc:1]: I believe this is the only module where we've docs for `placementGroupName`.

Comment (subhtk) [reply modules/machineset-aws-existing-placement-group.adoc:1]: Yeah. And will update in the release notes as well.

File: modules/machineset-aws-existing-placement-group.adoc [hunk 4]
```diff
@@ -55,20 +55,25 @@ spec:
...
+          placementGroupPartition: <placement partition number> # <6>
 # ...
 ----
 <1> Specify an instance type that link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html#efa-instance-types[supports EFAs].
 <2> Specify the `EFA` network interface type.
 <3> Specify the zone, for example, `us-east-1a`.
 <4> Specify the region, for example, `us-east-1`.
 <5> Specify the name of the existing AWS placement group to deploy machines in.
+[Optional]
+<6> Specify the partition number of the existing AWS placement group to deploys machines in.
```
Comment (huali9) [inline modules/machineset-aws-existing-placement-group.adoc:67]: Not sure if it needs to be stated that partition numbers can only be used on placement groups using the "partition" strategy

Comment (chiragkyal) [reply modules/machineset-aws-existing-placement-group.adoc:67]: It's good to add it explicitly, but in that case, there are other constraints as well, like partition number - which we would need to add. 

I see there is a note just above the spec, with the link to the AWS documentation. I think this should suffice with everything related to pre-requisite to use this field. WDYT?
```
Ensure that the link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/placement-groups.html#limitations-placement-groups[rules and limitations] for the type of placement group that you create are compatible with your intended use case.
```

Comment (huali9) [reply modules/machineset-aws-existing-placement-group.adoc]: Yeah, that should be good.

Comment (huali9) [reply modules/machineset-aws-existing-placement-group.adoc]: LGTM

File: modules/machineset-aws-existing-placement-group.adoc (hunk 1 above)
Comment (chiragkyal) [reply modules/machineset-aws-existing-placement-group.adoc:58]: @subhtk could you pls update it as well? Thanks!

File: modules/machineset-aws-existing-placement-group.adoc [hunk 5]
```diff
@@ -55,20 +55,24 @@ spec:
...
 <5> Specify the name of the existing AWS placement group to deploy machines in.
+<6> Optional: Specify the partition number of the existing AWS placement group to deploys machines in. 
 
 .Verification
 
 * In the AWS console, find a machine that the machine set created and verify the following in the machine properties:
 
 ** The placement group field has the value that you specified for the `placementGroupName` parameter in the machine set.
 
+** The placement group partition field has the value that you specified for the `placementGroupPartition` parameter in the machine set.
```
Comment (chiragkyal) [inline modules/machineset-aws-existing-placement-group.adoc:74]: nit: checked in the AWS console; the field name is `Partition number` 
```suggestion
** The partition number field has the value that you specified for the `placementGroupPartition` parameter in the machine set.
```

File: modules/machineset-aws-existing-placement-group.adoc (hunk 4 above)
Comment (subhtk) [reply modules/machineset-aws-existing-placement-group.adoc]: Great

**Comment by chiragkyal** [comment]
/lgtm
Thanks!

File: modules/machineset-aws-existing-placement-group.adoc [hunk 6]
```diff
@@ -55,20 +55,24 @@ spec:
...
           placementGroupName: <placement_group> # <5>
+          placementGroupPartition: <placement_group_partition_number> # <6>
 # ...
 ----
 <1> Specify an instance type that link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html#efa-instance-types[supports EFAs].
 <2> Specify the `EFA` network interface type.
 <3> Specify the zone, for example, `us-east-1a`.
 <4> Specify the region, for example, `us-east-1`.
 <5> Specify the name of the existing AWS placement group to deploy machines in.
+<6> Optional: Specify the partition number of the existing AWS placement group to deploys machines in.
```
Comment (skopacz1) [inline modules/machineset-aws-existing-placement-group.adoc:66]:
```suggestion
<6> Optional: Specify the partition number of the existing AWS placement group to deploy machines in. 
```

**Review by skopacz1 — COMMENTED** [review]
Just one tiny thing, otherwise LGTM!

**Review by jldohmann — APPROVED** [review]
lgtm, nice work!

**Pull request merged**